backend/src/cache/client_manager.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/emicklei/go-restful/log"
+	"github.com/golang/glog"
+	"github.com/jinzhu/gorm"
+	"github.com/kubeflow/pipelines/backend/src/cache/client"
+	"github.com/kubeflow/pipelines/backend/src/cache/model"
+
+	"github.com/kubeflow/pipelines/backend/src/cache/storage"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+const (
+	DBName                   = "cachedb"
+	DefaultConnectionTimeout = "6m"
+)
+
+type ClientManager struct {
+	db         *storage.DB
+	cacheStore storage.ExecutionCacheStoreInterface
+	time       util.TimeInterface
+}
+
+func (c *ClientManager) CacheStore() storage.ExecutionCacheStoreInterface {
+	return c.cacheStore
+}
+
+func (c *ClientManager) Close() {
+	c.db.Close()
+}
+
+func (c *ClientManager) init(params WhSvrDBParameters) {
+	timeoutDuration, _ := time.ParseDuration(DefaultConnectionTimeout)
+	db := initDBClient(params, timeoutDuration)
+
+	c.time = util.NewRealTime()
+	c.db = db
+	c.cacheStore = storage.NewExecutionCacheStore(db, c.time)
+
+}
+
+func initDBClient(params WhSvrDBParameters, initConnectionTimeout time.Duration) *storage.DB {
+	driverName := params.dbDriver
+	var arg string
+
+	switch driverName {
+	case mysqlDBDriverDefault:
+		arg = initMysql(params, initConnectionTimeout)
+	default:
+		glog.Fatalf("Driver %v is not supported", driverName)
+	}
+
+	// db is safe for concurrent use by multiple goroutines
+	// and maintains its own pool of idle connections.
+	db, err := gorm.Open(driverName, arg)
+	util.TerminateIfError(err)
+
+	db.DropTableIfExists(&model.ExecutionCache{})
```

Comment:
Should we drop the DB every time? Or is this for debugging only?

Reply:
No. This PR is still WIP and drop table here is for debugging.